### DISCUSSION TITLE
cannot find module 'tree-sitter-fpl'

### DISCUSSION THREAD

**Discussion by bookofproofs** [atom/atom#22635] [Q&A]
Hi folks,
I'm new to atom, tree-sitter, npm, etc., and facing a very steep learning curve. Please help.

I created an EBNF grammar for a new language I call the "Formal Proving Language" for expressing proof-based mathematics. Now, I want to integrate syntax highlighting for this language in a user-friendly editor. I heard that atom is "hackable" and supports syntax highlighting for new languages using tree-sitter. I read the [atom docs](https://flight-manual.atom.io/hacking-atom/sections/creating-a-grammar/). I accomplished the following so far are the following: 

1. I created a tree-sitter grammar for my language. 
2. I published my grammar at npm as ["tree-sitter-fpl"](https://www.npmjs.com/package/tree-sitter-fpl).
3. I created an initial git repository for the atom package (see [bookofproofs/language-fpl](https://github.com/bookofproofs/language-fpl)).
4. I published the atom package (see [language-fpl](https://atom.io/packages/language-fpl)).

Now, when I start atom (version 1.57.0), I get the error

`Failed to load a language-fpl package grammar Cannot find module 'tree-sitter-fpl.'`

What am I doing wrong, and why does atom have problems with finding the module?

**Comment by ThatXliner** [comment]
Try running `npm install tree-sitter-fpl` on your package's directory

**Reply by bookofproofs** [reply]
Thank you, but this leads to 

`Refusing to install package with name "tree-sitter-fpl" under a package
npm ERR! also called "tree-sitter-fpl". Did you name your project the same
npm ERR! as the dependency you're installing?`

**Reply by bookofproofs** [reply]
I suppose, there is a problem with the release version, I'll drop everything and try again. Thank you for your help!

**Reply by ThatXliner** [reply]
I meant the Atom package

**Reply by icecream17** [reply]
did you mean `apm install`?

**Reply by bookofproofs** [reply]
`apm install` said `Installing modules done`. The atom error message `Failed to load a language-fpl package grammar Cannot find module 'tree-sitter-fpl.'` did not disappear, even after restarting atom. I suppose that the error message is misleading. I'll look at the packages using tree-sitter for other languages. Maybe I missed something.

**Reply by icecream17** [reply]
Er... `apm install tree-sitter-fpi`?
